src/ucp/tag/tag_match.inl

```diff
@@ -67,7 +73,7 @@ ucp_tag_match_calc_hash(ucp_tag_t tag)
 static UCS_F_ALWAYS_INLINE ucs_queue_head_t*
 ucp_tag_exp_get_queue_for_tag(ucp_tag_match_t *tm, ucp_tag_t tag)
 {
-    return &tm->expected.hash[ucp_tag_match_calc_hash(tag)];
+    return &tm->expected.hash[ucp_tag_match_calc_hash(tag)].queue;
```

Comment:
imho better change this function to return ucp_request_queue_t *, instead of doing container_of later when posting offload

Reply:
In this case I'd need to update all 3 `..get_exp..` functions (and the places where they are used), because they depend on each other. This would look worse. Should I go for it?

Reply:
i think so - because this PR extended the "queue" object, so now the new extended queue should be returned, mostly to avoid up-casting [ucs_deriver_of] later on, without any real "polymorphism"